src/XMakeTasks/FileIO/WriteLinesToFile.cs

```diff
@@ -83,12 +85,12 @@ public override bool Execute()
                     }
                 }
 
-                Encoding encode = null;
+                Encoding encoding = s_defaultEncoding;
                 if (_encoding != null)
```

Comment:
Making the `s_defaultEncoding` field with `readonly` allows you to avoid this logic.